### DISCUSSION TITLE
rebuild code-server question

### DISCUSSION THREAD

**Discussion by slion-L** [coder/code-server#5583] [Q&A]
i run `yarn watch` and success. but it seems not same with the release version. there is not `bin/code-server`. it that correct? how can i restart code-server?

**Comment by jsjoeio** [comment]
`yarn watch` will watch for changes and restart automatically. i.e. if you hit save or update a file in `/src` then it should restart automatically

**Reply by slion-L** [reply]
yes, it start automatically. in the release version, i can start code-server by `./bin/code-server`. i just can't find the `./bin/code-server` after i build it.

**Reply by slion-L** [reply]
there is another question atfer i run `yarn watch`. my rebuilding version shows `Forbidden.` after login. it seems related to the code below. how can i do to fix it?
![image](https://user-images.githubusercontent.com/53867313/192089114-2d7a1cc1-9e7a-47ee-9a4a-676f1c1296e6.png)

**Reply by jsjoeio** [reply]
> after i build it.

Which command are you using to build it? Just want to make sure I point you to the right place.

Ah, sounds like you forgot to run `quilt push -a`. See instructions [here](https://coder.com/docs/code-server/latest/CONTRIBUTING#development-workflow) for using `yarn watch` and developing locally.
